drivers/vulkan/rendering_device_driver_vulkan.cpp

```diff
@@ -2991,7 +2989,7 @@ Error RenderingDeviceDriverVulkan::swap_chain_resize(CommandQueueID p_cmd_queue,
 	swap_create_info.minImageCount = desired_swapchain_images;
 	swap_create_info.imageFormat = swap_chain->format;
 	swap_create_info.imageColorSpace = swap_chain->color_space;
-	swap_create_info.imageExtent = native_display_size;
+	swap_create_info.imageExtent = extent;
```

Comment:
Line 2901 would handle the case of UINT32_MAX extents from GetPhysicalDeviceSurfaceCapabilitiesKHR that means "indicating that the surface size will be determined by the extent of a swapchain targeting the surface".  This would be stored to native_display_size but then not updated when the others are handled.  

Since native_display_size wasn't used anywhere else it was simple to just use the calculated extents instead.